### PR TITLE
requiresNoCD->requiresPatch

### PR DESCRIPTION
**Pull request by ImperatorS79**
name should be the same than folder name

Again sorry for this

### CODE REVIEW

**Comment by plata** [comment]
It was correct before: <https://github.com/PlayOnLinux/Scripts/wiki/application.json>

**Comment by ImperatorS79** [comment]
I was actually looking for a way to solve my apps tab problem, and I find it...
The wiki should be updated due to 4 level JS namespace and requiresNoCD->requiresPatches

**Comment by ImperatorS79** [comment]
To be sure before using : what type of checksum are you using in Phoenicis ?

**Comment by ImperatorS79** [comment]
I tried to run the script, and got this error : 

Script not found : [Engines, Wine, Verbs, d3dx9]

org.phoenicis.scripts.interpreter.ScriptException: Script not found: [Engines,  Wine, Verbs, d3dx9]
	at org.phoenicis.scripts.interpreter.ScriptFetcher.getScript(ScriptFetcher.java:38)
	at org.phoenicis.scripts.interpreter.ScriptFetcher.getScript(ScriptFetcher.java:45)
	at org.phoenicis.scripts.nashorn.builtins.IncludeInjector.lambda$injectInto$0(IncludeInjector.java:30)
	at jdk.nashorn.internal.scripts.Script$133$\^eval\_.:program(<eval>:2)
	at jdk.nashorn.internal.runtime.ScriptFunctionData.invoke(ScriptFunctionData.java:637)
	at jdk.nashorn.internal.runtime.ScriptFunction.invoke(ScriptFunction.java:494)
	at jdk.nashorn.internal.runtime.ScriptRuntime.apply(ScriptRuntime.java:393)
	at jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:449)
	at jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:406)
	at jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:402)
	at jdk.nashorn.api.scripting.NashornScriptEngine.eval(NashornScriptEngine.java:155)
	at javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:264)
	at org.phoenicis.scripts.nashorn.NashornEngine.eval(NashornEngine.java:45)
	at org.phoenicis.scripts.nashorn.NashornScriptInterpreter.runScript(NashornScriptInterpreter.java:35)
	at org.phoenicis.scripts.interpreter.BackgroundScriptInterpreter.lambda$runScript$0(BackgroundScriptInterpreter.java:35)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

**Comment by plata** [comment]
I have updated the wiki to reflect the `requiresPatch` and 4 level namespace changes.

The checksum is `SHA-1` by default (`SHA` for [MessageDigest](https://docs.oracle.com/javase/8/docs/api/java/security/MessageDigest.html)). But you can also specify a different algorithm via e.g. `Downloader#algorithm("MD5")`.

File: Applications/Games/STAR WARS- Empire at War - Gold Pack/application.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
-	"name":	"STAR WARS™ Empire at War - Gold Pack",
+	"name":	"STAR WARS- Empire at War - Gold Pack",
```
Comment (plata) [inline Applications/Games/STAR WARS- Empire at War - Gold Pack/application.json]: You should leave it like it has been before.
Also I just noticed that the folder is called "STAR WARS- Empire at War - Gold Pack". Looks a bit strange without space before "-". Could you change it to "STAR WARS Empire at War - Gold Pack" ?

**Comment by plata** [comment]
Regarding the script not found: seems like you have a " " in front of your "Wine".

**Comment by ImperatorS79** [comment]
Although the patch is downloaded and seems to be executed, it isn't.  I verified with POL4, and a a window should appear. Here, it doesn"t

File: Applications/Games/STAR WARS - Empire at War - Gold Pack/Local/script.js [hunk 2]
```diff
@@ -16,11 +16,11 @@ new LocalInstallerScript()
         new Downloader()
             .wizard(wizard)
             .url("http://static.dolimg.com/mh_netstorage/lucasfilm/patches/pc/EAW_RAM_MPLobby_update.exe")
-            .checksum("9fdecb9609a17f7358f2984a3278b484")
+            .checksum("63233107fab4c58413df04ce1d07fe65e7145329")
             .to(wine.prefixDirectory + "drive_c/users/Public/Documents/EAW_RAM_MPLobby_update.exe")
             .get();
         
-        wine.runInsidePrefix(wine.prefixDirectory + "drive_c/users/Public/Documents/EAW_RAM_MPLobby_update.exe","");
+        wine.run(wine.prefixDirectory + "drive_c/users/Public/Documents/EAW_RAM_MPLobby_update.exe","");
```
Comment (plata) [inline Applications/Games/STAR WARS - Empire at War - Gold Pack/Local/script.js]: I think you can leave away the second argument ("" should be default). Any idea why `runInsidePrefix` doesn't work?

Comment (ImperatorS79) [inline Applications/Games/STAR WARS - Empire at War - Gold Pack/Local/script.js]: No, I don't.

Comment (plata) [inline Applications/Games/STAR WARS - Empire at War - Gold Pack/Local/script.js]: I just checked again. `runInsidePrefix` is defined as `run(this.prefixDirectory + "/drive_c/" + executable, args)`. So in your case you get prefix + "/drive_c/" + prefix + "/drive_c/users...". If you write `runInsidePrefix("users/Public...")`, it should work.
Can you try that? I will merge this afterwards.

Comment (ImperatorS79) [inline Applications/Games/STAR WARS - Empire at War - Gold Pack/Local/script.js]: Ok, now everything is working fine.

**Pull request merged**